prompb/remote.proto

```diff
@@ -23,10 +23,34 @@ message WriteRequest {
   repeated prometheus.TimeSeries timeseries = 1 [(gogoproto.nullable) = false];
 }
 
+// ReadRequest represents a remote read request.
 message ReadRequest {
   repeated Query queries = 1;
+
+  enum ResponseType {
+    // Server will return a ReadResponse with raw samples.
+    //
+    // Response headers:
+    // Content-Type: "application/x-protobuf"
+    // Content-Encoding: "snappy"
+    SAMPLES = 0;
+
+    // Server will stream a varint delimited ChunkedReadResponse message that contains XOR encoded chunks for a single series.
+    //
+    // Response headers:
+    // Content-Type: "application/x-streamed-protobuf; proto=prometheus.ChunkedReadResponse"
+    // Content-Encoding: ""
+    STREAMED_XOR_CHUNKS = 1;
+  }
+
+  // accepted_response_types allows negotiating the content type of the response.
+  // Response types are taken from the list in the FIFO order. If no response type is implemented by server, error is returned.
+  // For servers that do not support this field, the SAMPLES response type is used strictly.
+  // If field is empty, the server should assume SAMPLES response type is accepted by the caller.
+  repeated ResponseType accepted_response_types = 2;
```

Comment:
Given that empty implies SAMPLES, and we assume all clients must accept a SAMPLES response, do we really need the SAMPLES enum value?